HarpGuru/HarpFace/HarpCell/useQuizCycle/index.ts

```diff
@@ -0,0 +1,47 @@
+import { useGlobal, useDispatch } from 'reactn'
+import { useEffect } from 'react'
+import { IsActiveIds } from 'harpstrata'
+
+import { usePositionAnalysis } from '../usePositionAnalysis'
+import { YXCoord } from '../types'
+import { ExperienceModes } from '../../../helpers/setGlobalReactNState'
+import { usePrevious } from '../../../helpers'
+
+export const useQuizCycle = (yxCoord: YXCoord): void => {
+  const [activeExperienceMode] = useGlobal('activeExperienceMode')
+  const [quizQuestion] = useGlobal('quizQuestion')
+  const { thisIsActiveId } = usePositionAnalysis(yxCoord)
+  const previousIsActiveId = usePrevious(thisIsActiveId, IsActiveIds.Inactive)
+
+  const requestNextQuestion = useDispatch('requestNextQuestion')
+  const revealAnswer = useDispatch('revealAnswer')
+  useEffect(() => {
+    if (
+      thisIsActiveId === IsActiveIds.Active &&
+      previousIsActiveId === IsActiveIds.Inactive &&
+      activeExperienceMode === ExperienceModes.Quiz
+    ) {
+      const revealTimer = setTimeout(() => {
+        // TODO: why does the test fail if I don't test for
+        // requestNextQuestion first? Is it just because I
+        // haven't mocked that global reducer in the test?
+        if (revealAnswer) {
+          revealAnswer()
+        }
+      }, 200)
+      const nextQuestionTimer = setTimeout(() => {
+        // TODO: why does the test fail if I don't test for
+        // requestNextQuestion first? Is it just because I
+        // haven't mocked that global reducer in the test?
+        if (requestNextQuestion) {
+          requestNextQuestion()
+        }
+      }, 1500)
+      return () => {
+        clearTimeout(revealTimer)
+        clearTimeout(nextQuestionTimer)
+      }
+    }
+    return undefined
+  }, [quizQuestion, thisIsActiveId])
+}
```

Comment:
I think this could be moved in the future and it could result in performance improvements. I need to note this down in an optimisation issue.

Reply:
Referenced in open issue: https://github.com/js-jslog/harpnative/issues/52#issuecomment-663847598